src/main/java/com/celements/mandatory/AddDateIDXAsIndexToRecycleBin.java

```diff
@@ -0,0 +1,45 @@
+package com.celements.mandatory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.Requirement;
+
+import com.celements.query.IQueryExecutionServiceRole;
+import com.xpn.xwiki.XWikiException;
+
+@Component("celements.mandatory.addIndexToRecycleBin")
+public class AddDateIDXAsIndexToRecycleBin implements IMandatoryDocumentRole {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AddDateIDXAsIndexToRecycleBin.class);
+
+  @Requirement
+  IQueryExecutionServiceRole queryExecService;
+
+  @Override
+  public List<String> dependsOnMandatoryDocuments() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public void checkDocuments() throws XWikiException {
+    LOGGER.info("executing AddDateIDXAsIndexToRecycleBin");
+    // FIXME make sure that these operations are ONLY executed ONCE on any DB.
```

Comment:
please fix this `FIXME` instead of producing more errors :) check the following sql:
```
select CONSTRAINT_NAME from INFORMATION_SCHEMA.KEY_COLUMN_USAGE
where TABLE_SCHEMA = 'prefix_db'
  and TABLE_NAME = 'xwikirecyclebin'
  and CONSTRAINT_NAME = 'dateIDX';
```
with it you can find out if the index already exists. In `IQueryExecutionServiceRole` provide a `boolean existsConstraint(String database, String table, String name)`.